src/chart_types/partition_chart/state/selectors/get_highlighted_shapes.ts

```diff
@@ -19,21 +19,58 @@
 
 import createCachedSelector from 're-reselect';
 
+import { LegendPath } from '../../../../state/actions/legend';
 import { GlobalChartState } from '../../../../state/chart_state';
 import { getChartIdSelector } from '../../../../state/selectors/get_chart_id';
-import { QuadViewModel } from '../../layout/types/viewmodel_types';
+import { getSettingsSpecSelector } from '../../../../state/selectors/get_settings_specs';
+import { DataName, QuadViewModel } from '../../layout/types/viewmodel_types';
 import { partitionGeometries } from './geometries';
 
-const getHighlightedLegendItemKey = (state: GlobalChartState) => state.interactions.highlightedLegendItemKey;
+const getHighlightedLegendItemPath = (state: GlobalChartState) => state.interactions.highlightedLegendPath;
+
+const legendStrategies = Object.freeze({
+  node: (legendPath: LegendPath) => ({ path }: { path: LegendPath }) =>
```

Comment:
Can we add types here? Also, is `freeze` needed?

```suggestion
type LegendStrategyFn = (legendPath: LegendPath) => (shape: { path: LegendPath; dataName: DataName }) => boolean;
const legendStrategies: Record<LegendStrategy, LegendStrategyFn> = Object.freeze({
  node: (legendPath: LegendPath) => ({ path }: { path: LegendPath }) =>
```

Reply:
Thanks, good catch and recommendation, I also shed the per object value type info with this
a53dd28395a05eafe52ebd09e8b1e2563ec5c6ee 
5ec5535d7a53af28250d716631619b6e4f3e0be1